flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/IntervalFreshness.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.util.Objects;
+
+/**
+ * The {@link IntervalFreshness} represents freshness definition of {@link
+ * CatalogMaterializedTable}. It encapsulates the string interval value along with time unit,
+ * allowing for flexible representation of different freshness type. Moreover, it can provide
+ * detailed raw information for some specific operations.
+ */
+@PublicEvolving
+public class IntervalFreshness {
+
+    private final String interval;
+    private final TimeUnit timeUnit;
+
+    public IntervalFreshness(String interval, TimeUnit timeUnit) {
```

Comment:
I'm curious why it's a String here. If it were an int, we could validate the legality of the interval before converting it into IntervalFreshness.

Reply:
Considering future extensibility, it may support other types of time units, such as DAY TO HOUR, DAY TO SECOND, which are not directly convertible to numeric types, so here it is defined as String type first, which is the widest type.

1. https://calcite.apache.org/docs/reference.html